### ISSUE TITLE
@JsonView: support more than one view

### ISSUE THREAD

**Issue by renaudcerrato** [FasterXML/jackson-jaxrs-providers#84]
Currently, `@JsonView` annotation on endpoint only use the first view passed to it.

**Comment by cowtowncoder** [comment]
I am not sure I understand this: how would defining of multiple views work? Jackson requires a single view to be used for serialization, so endpoints can only similarly define one active view to use via annotation. So what would be your use case; what are you trying to achieve?

**Comment by renaudcerrato** [comment]
Mmmm... I see, that make sense. What I'm trying to achieve: my `User` model have an `email` field. I'd like to somewhat obfuscate that field on _public view_ only:

```
{
    "firstname": "John",
    "lastname": "Doh",
    "email": "j**n***h@g**i**o*
}
```

The above could probably be easily achieved by using a custom serializer but since I'm in a jax-rs resource, I don't want to have to deal with my `ObjectMapper` directly. 

Any ideas?

**Comment by cowtowncoder** [comment]
@renaudcerrato Ah yes. So dynamically changing the active view, based on user.

This is actually why a new (#33 included in 2.3.0...) feature was added to allow for configuring `ObjectReader` and/or `ObjectWriter` used. Classes you'll need are:

* `ObjectWriterInjector`, `ObjectWriterModifier`

and test at `json/src/test/java/com/fasterxml/jackson/jaxrs/json/dw/WriteModificationsTestBase.java` might be useful. I wish there was better documentation, but maybe that'll get you started.

For what it's worth, changing view was the reason I added this functionality; definitely makes sense to be able to choose it, but seemed very difficult to be able to provide such changeability using pure Annotation based approach.
